packages/datadog-instrumentations/src/langchain.js

```diff
@@ -46,51 +16,14 @@ for (const extension of extensions) {
     file: `dist/language_models/chat_models.${extension}`,
     versions: ['>=0.1']
   }, exports => {
-    const BaseChatModel = exports.BaseChatModel
-    shimmer.wrap(
-      BaseChatModel.prototype,
-      'generate',
-      generate => wrapLangChainPromise(generate, 'chat_model')
-    )
     return exports
   })
 
   addHook({ name: '@langchain/core', file: `dist/language_models/llms.${extension}`, versions: ['>=0.1'] }, exports => {
-    const BaseLLM = exports.BaseLLM
-    shimmer.wrap(BaseLLM.prototype, 'generate', generate => wrapLangChainPromise(generate, 'llm'))
     return exports
   })
 
   addHook({ name: '@langchain/core', file: `dist/embeddings.${extension}`, versions: ['>=0.1'] }, exports => {
-    // we cannot patch the prototype of the Embeddings class directly
-    // this is because the "abstract class Embeddings" is transpiled from TypeScript to not include abstract functions
-    // thus, we patch the exported class directly instead instead.
-
-    shimmer.wrap(exports, 'Embeddings', Embeddings => {
```

Comment:
Removing the JS implementation will make this stop working in Node <20.6

Reply:
Done.